packages/react-devtools-extensions/src/parseHookNames.js

```diff
@@ -109,50 +145,52 @@ export default async function parseHookNames(
       throw Error('Hook source code location not found.');
     }
 
-    const fileName = hookSource.fileName;
-    if (fileName == null) {
-      throw Error('Hook source code location not found.');
-    } else {
-      if (!fileNameToHookSourceData.has(fileName)) {
-        const hookSourceData: HookSourceData = {
-          hookSource,
-          originalSourceAST: null,
-          originalSourceCode: null,
-          runtimeSourceCode: null,
-          sourceConsumer: null,
-          sourceMapURL: null,
-          sourceMapContents: null,
-        };
-
-        // If we've already loaded source/source map info for this file,
-        // we can skip reloading it (and more importantly, re-parsing it).
-        const metadata = fileNameToMetadataCache.get(fileName);
-        if (metadata != null) {
-          if (__DEBUG__) {
-            console.groupCollapsed(
-              'parseHookNames() Found cached metadata for file "' +
-                fileName +
-                '"',
-            );
-            console.log(metadata);
-            console.groupEnd();
-          }
+    const runtimeSourceURL = hookSource.fileName;
 
-          hookSourceData.originalSourceAST = metadata.originalSourceAST;
-          hookSourceData.originalSourceCode = metadata.originalSourceCode;
-          hookSourceData.sourceConsumer = metadata.sourceConsumer;
+    const locationKey = getLocationKey(hookSource);
+    if (!locationKeyToHookSourceData.has(locationKey)) {
+      if (runtimeSourceURL == null) {
+        // Not reachable because getLocationKey will have thrown
+        throw Error('Hook source code location not found.');
+      }
+      const hookSourceData: HookSourceData = {
+        hookSource,
+        runtimeSourceURL,
+        originalSourceAST: null,
+        originalSourceCode: null,
+        runtimeSourceCode: null,
+        sourceConsumer: null,
+        sourceMapURL: null,
+        originalSourceURL: null,
+      };
+
+      // If we've already loaded the source map info for this file,
+      // we can skip reloading it (and more importantly, re-parsing it).
+      const runtimeMetadata = runtimeURLToMetadataCache.get(
+        hookSourceData.runtimeSourceURL,
+      );
+      if (runtimeMetadata != null) {
+        if (__DEBUG__) {
+          console.groupCollapsed(
+            'parseHookNames() Found cached runtime metadata for file "' +
+              hookSourceData.runtimeSourceURL +
+              '"',
+          );
+          console.log(runtimeMetadata);
+          console.groupEnd();
         }
-
-        fileNameToHookSourceData.set(fileName, hookSourceData);
+        hookSourceData.sourceConsumer = runtimeMetadata.sourceConsumer;
       }
+
```

Comment:
We may want to add another cache read path here, in the special case where:

1. there is no source map
2. therefore `originalSourceURL` will be equal to `runtimeSourceURL`
3. therefore, potentially, `originalURLToMetadataCache.has(runtimeSourceURL)`.
   * The TODO comment in `parseSourceAST`, about distinguishing between runtime URLs and true source mapped URLs in this cache, complicates this - but only slightly.

In that case we can directly grab `originalSourceAST` from `originalURLToMetadataCache`, and we may not need to ever fetch `runtimeSourceCode`. (I know we pass it to `astUtils`, but it actually seems unused there.)

Reply:
I think this can be done in a follow up.